L72-core-google_default_credentials-extension.md

```diff
@@ -0,0 +1,58 @@
+Allow Call Credentials to be Specified in `grpc_google_default_credentials_create`
+----
+* Author(s): rbellevi
+* Approver: markdroth
+* Status: Draft
+* Implemented in: Core
+* Last updated: July 6th, 2020
+* Discussion at: https://groups.google.com/g/grpc-io/c/fZNm4pU8e3s/m/2Be8u1n7BQAJ
+
+## Abstract
+
+This document proposes that `grpc_google_default_credentials_create` be
+amended to allow the user to specify their desired call credentials.
+
+## Background
+
+The Google default credentials created by the
+`grpc_google_default_credentials_create` function in Core enable connection to
+Google services via a combination of ALTS and SSL credentials, along with a special oauth2
+token that must assert the same identity as the channel-level ALTS credential.
+
+This oauth2 token attached by `grpc_google_default_credentials_create` is based
+on the default VM service account, but the auth libraries for wrapped languages
+allow the creation of channels with non-default service accounts. In this case,
+a mismatch in identity occurs that may cause an unexpected persistent failure of
+RPCs at runtime.
+
+While the default service account is the correct option is most cases, there
```

Comment:
I think we might want to change wording here. This is not so much about allowing higher layers to authenticate alternative identities, rather than it is allowing client libraries to continue to use their own mechanisms to get per-RPC tokens, when potentially using ALTS. I.e. this API described here allows  client libraries to continue uses the auth metadata plugin APIs to get per-RPC metadata, when using ALTS.